hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/checkpoint/KafkaConnectHdfsProvider.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utilities.sources.checkpoint;
+
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Generate checkpoint from Kafka-Connect-HDFS.
+ */
+public class KafkaConnectHdfsProvider implements CheckPointProvider {
```

Comment:
How do you plan to integrate this class with DeltaStreamer?

Reply:
few different options IMO:

- pass the output of this tool as `--checkpoint` to delta streamer in the first run
- integrate this tool with HoodieWriteClient or Spark writer or HdfsImporter as an option controlled by config, user can save the checkpoint when they are not using delta streamer. so next time they can switch seamlessly.

Reply:
I am more inclined towards integrating this tool at proper places in our code base rather than specifying the checkpoint manually. You plan to do this integration in a separate PR?

Reply:
yes, prefer to keep the PR small enough for review. WDYT?

Reply:
Sounds good to me. @vinothchandar to take the final call. :)

Reply:
I think we can integrate this using a new `--initial-checkpoint-provider` flag that takes the name of the class that provides an initial checkpoint, if no commits/checkpoints are found on the hudi dataset itself.. The current behavior is probably to pass an Optional.empty to the Source class and let the sources decide (for e,g kafka will start from earliest/latest depending on what the user wants it to be)? We can encapsulate this behavior in a another InitialCheckpointProvider implementation, which will be the default value for `--initial-checkpoint-provider` flag

Reply:
Do we wanna integrate this with the delta streamer? 
My idea was to use this tool when not using delta streamer and save a checkpoint into the commit. Then the user can switch to delta streamer in the next run. 
If we integrate this with delta streamer, let's say the users are using kafka connect, they need to use Spark Datasource reader to do the initial bulk insert first, then the second run will be using delta streamer. But if we do the checkpoint scan in the second run, the checkpoint will not represent the data in the Hudi dataset, which means we need to do the checkpoint scan in the first run. It that makes sense? Or I am missing something

Reply:
renamed to `InitialCheckPointProvider`